src/Tribe/RSVP.php

```diff
@@ -2170,10 +2576,14 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
 		$qty = max( $ticket_qty, 0 );
 
 		// Throw an error if Qty is bigger then Remaining
-		if ( $ticket_type->managing_stock() && $qty > $ticket_type->inventory() ) {
-			$url = add_query_arg( 'rsvp_error', 2, get_permalink( $post_id ) );
-			wp_redirect( esc_url_raw( $url ) );
-			tribe_exit();
+		if ( $ticket_type->managing_stock() && $ticket_type->inventory() < $qty ) {
+			if ( $redirect ) {
+				$url = add_query_arg( 'rsvp_error', 2, get_permalink( $post_id ) );
+				wp_redirect( esc_url_raw( $url ) );
```

Comment:
* :warning: wp_redirect() found. Using wp_safe_redirect(), along with the allowed_redirect_hosts filter if needed, can help avoid any chances of malicious redirects within code. It is also important to remember to call exit() after a redirect so that no other unwanted code is executed.
* :no_entry_sign: `wp_redirect()` should almost always be followed by a call to `exit;`.

<sub>hash: 42d76fcdde7c9f455a97c46137422ebb</sub>